.github/actions/support-package-driver-matrix/driver-matrix.py

```diff
@@ -0,0 +1,149 @@
+#       "kmod": "upstream"
+#     },
+#     "2.4.0": {
+#       "ebpf": "unavailable",
+#       "kmod": "upstream"
+#     },
+#     "2.5.0": {
+#       "ebpf": "unavailable"
+#     }
+#   },
+#   "4.18.0-80.el8.x86_64": {
+#     "2.2.0": {
+#       "kmod": "upstream",
+#       "ebpf": "upstream"
+#     },
+#     "2.3.0": {
+#       "kmod": "upstream",
+#       "ebpf": "upstream"
+#     },
+#     "2.4.0": {
+#       "kmod": "upstream",
+#       "ebpf": "upstream"
+#     },
+#     "2.5.0": {
+#       "ebpf": "upstream"
+#     }
+#   },
+#   ...
+# }
+#
+# If you are interested on finding a specific kernel, you can use jq with a
+# filter similar to the following:
+#   jq '."6.2.8-300.fc38.x86_64"' driver-matrix.json
+#   {
+#     "2.4.0": {
+#       "kmod": "upstream",
+#       "ebpf": "upstream"
+#     },
+#     "2.5.0": {
+#       "ebpf": "upstream"
+#     }
+#   }
+
+import argparse
+import json
+import os
+import sys
+import re
+
+ebpf_re = re.compile(r'(\d+\.\d+\.\d+)/collector-ebpf-(\d+\.\d+\.\d+.*)\.o\.gz$')
+kmod_re = re.compile(r'(\d+\.\d+\.\d+)/collector-(\d+\.\d+\.\d+.*)\.ko\.gz$')
+unavailable_re = re.compile(r'(\d+\.\d+\.\d+)/\.collector-ebpf-(\d+\.\d+\.\d+.*)\.unavail$')
+
+
+def update_kernel(kernel_list: dict,
+                  kernel_version: str,
+                  driver_version: str,
+                  driver_type: str,
+                  status: str):
+    kernel = kernel_list.get(kernel_version)
+    if kernel is None:
+        kernel_list[kernel_version] = {
+            driver_version: {
+                driver_type: status,
+            }
+        }
+    else:
+        dver = kernel.get(driver_version)
+        if dver is None:
+            kernel[driver_version] = {
+                driver_type: status,
+            }
+        else:
+            dver[driver_type] = status
+
+
+def process_line(kernels: dict,
+                 available: str,
+                 line: str):
+    match = ebpf_re.search(line)
+    if match:
+        driver_version = match[1]
+        kernel_version = match[2]
+        update_kernel(kernels, kernel_version,
+                      driver_version, 'ebpf', available)
+        return
+
+    match = kmod_re.search(line)
+    if match:
+        driver_version = match[1]
+        kernel_version = match[2]
+        update_kernel(kernels, kernel_version,
+                      driver_version, 'kmod', available)
+        return
+
+    match = unavailable_re.search(line)
+    if match:
+        driver_version = match[1]
+        kernel_version = match[2]
+        update_kernel(kernels, kernel_version,
+                      driver_version, 'ebpf', 'unavailable')
+        return
+
+    print('Did not match any known drivers')
+
+
+def main(kernels: dict, file, available: str):
+    for line in sys.stdin:
+        line = line.rstrip()
+        process_line(kernels, available, line)
+
+    if file:
+        json.dump(kernels, file)
+    else:
+        print(json.dumps(kernels))
```

Comment:
we can use json.dump in both branches here - 

```
if file:
    json.dump(kernel, file)
else:
    json.dump(kernel, sys.stdout)
```

Or better still, we can pass in either file or stdout in the caller and then we don't need to care where we're dumping